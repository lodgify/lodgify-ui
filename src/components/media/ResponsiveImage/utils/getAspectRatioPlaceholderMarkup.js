import React from 'react';

import { getAspectRatioAsPercentage } from './getAspectRatioAsPercentage';
/**
 * @param  {number} imageWidth
 * @param  {number} imageHeight
 * @return {Node|undefined}
 */
export const getAspectRatioPlaceholderMarkup = (imageWidth, imageHeight) => {
  if (!!imageWidth && !!imageHeight) {
    return (
      <div
        style={{
          paddingBottom: `${getAspectRatioAsPercentage(
            imageWidth,
            imageHeight
          )}%`,
        }}
      />
    );
  }
};
