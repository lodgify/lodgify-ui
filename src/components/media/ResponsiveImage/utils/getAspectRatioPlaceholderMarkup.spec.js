import React from 'react';
import { mount } from 'enzyme';

import { getAspectRatioPlaceholderMarkup } from './getAspectRatioPlaceholderMarkup';

const getMarkup = (imageWidth, imageHeight) =>
  mount(<div>{getAspectRatioPlaceholderMarkup(imageWidth, imageHeight)}</div>);

describe('`getAspectRatioPlaceholderMarkup`', () => {
  describe('by default', () => {
    it('should return the right structure', () => {
      const actual = getMarkup(1000, 200);

      expect(actual).toMatchSnapshot();
    });
  });

  describe('if `imageWidth === undefined`', () => {
    it('should return the right structure', () => {
      const actual = getMarkup(undefined, 200);

      expect(actual).toMatchSnapshot();
    });
  });

  describe('if `imageHeight === undefined`', () => {
    it('should return the right structure', () => {
      const actual = getMarkup(1000, undefined);

      expect(actual).toMatchSnapshot();
    });
  });

  describe('if both arguments are undefined', () => {
    it('should return the right structure', () => {
      const actual = getMarkup(undefined, undefined);

      expect(actual).toMatchSnapshot();
    });
  });
});
