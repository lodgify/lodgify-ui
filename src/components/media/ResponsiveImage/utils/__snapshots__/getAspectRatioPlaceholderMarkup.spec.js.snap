// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`getAspectRatioPlaceholderMarkup\` by default should return the right structure 1`] = `
<div>
  <div
    style={
      Object {
        "paddingBottom": "20%",
      }
    }
  />
</div>
`;

exports[`\`getAspectRatioPlaceholderMarkup\` if \`imageHeight === undefined\` should return the right structure 1`] = `<div />`;

exports[`\`getAspectRatioPlaceholderMarkup\` if \`imageWidth === undefined\` should return the right structure 1`] = `<div />`;

exports[`\`getAspectRatioPlaceholderMarkup\` if both arguments are undefined should return the right structure 1`] = `<div />`;
